src/edata.cpp

```diff
@@ -409,6 +416,9 @@ void ConditionContext::applyCondition(const ExpData *edata,
                                  "not match ExpData (%zd).",
                                  model_->nk(), edata->fixedParameters.size());
           model_->setFixedParameters(edata->fixedParameters);
+          if(!edata->reinitializeFixedParameterInitialStates)
```

Comment:
I am confused, don't we want to check for `edata->reinitializeFixedParameterInitialStates` here?

Reply:
The thing is, that if `edata->reinitializeFixedParameterInitialStates == true`, then this has been set on Model a couple of lines before. This will populate the index array in Model. If we would execute the next line, this would overwrite them with an empty array.
Not great, but keeps backward compatibility for the moment and still allow selective reinitialization. To be cleaned up soon.